src/widgets/source/ui/Heatmap2DWidget.cpp

```diff
@@ -146,15 +146,16 @@ namespace SmartPeak
           {
             bool is_hovered = false;
             ImPlotPoint plot_point;
+            auto window_size = ImGui::GetWindowSize();
             ImPlot::SetColormap(ImPlotColormap_Jet);
-            ImPlot::ShowColormapScale(heatmap_data_.feat_value_min_, heatmap_data_.feat_value_max_, height_ - 70);
+            ImPlot::ShowColormapScale(heatmap_data_.feat_value_min_, heatmap_data_.feat_value_max_, window_size.y - 70);
             float available_width = ImGui::GetContentRegionAvailWidth();
             ImGui::SameLine();
             const ImPlotFlags imPlotFlags = ImPlotAxisFlags_LockMin | ImPlotAxisFlags_LockMax | ImPlotAxisFlags_TickLabels;
             if (ImPlot::BeginPlot(plot_title_.c_str(),
               heatmap_data_.feat_heatmap_x_axis_title.c_str(),
               heatmap_data_.feat_heatmap_y_axis_title.c_str(),
-              ImVec2(available_width - 80, height_ - 70),
+              ImVec2(available_width - 80, window_size.y - 70),
```

Comment:
It would be nice to derive the constant values automatically whenever possible. If not it would be very useful to declare them with descriptive names or document what they are for.

Reply:
I agree. However, i will create another task for that, as this is not something introduced in that PR and is out of the scope of this feature for now.

Reply:
I agree. I have created a separate task for that on JIRA (as this PR didn't introduce that, and it would be out of scope to fix it now).